actors/builtin/multisig/multisig_actor_state.go

```diff
@@ -25,9 +25,8 @@ func (st *MultiSigActorState) AmountLocked(elapsedEpoch abi.ChainEpoch) abi.Toke
 		return abi.NewTokenAmount(0)
 	}
 
-	// FIXME: This division is probably incorrect, see issue: https://github.com/filecoin-project/specs-actors/issues/32
-	lockedProportion := (st.UnlockDuration - elapsedEpoch) / st.UnlockDuration
-	return big.Mul(st.InitialBalance, big.NewInt(int64(lockedProportion)))
+	unitLocked := big.Div(st.InitialBalance, big.NewInt(int64(st.UnlockDuration)))
+	return big.Mul(unitLocked, big.Sub(big.NewInt(int64(st.UnlockDuration)), big.NewInt(int64(elapsedEpoch))))
```

Comment:
This goes negative as elapsedEpoch increase above unlockDuration. Pull out the subtraction and set a floor at 0.

Reply:
that is caught by the return statement at L25